db/migrations/1657379568_create_courses_table.up.sql

```diff
@@ -2,9 +2,9 @@ BEGIN;
 
 CREATE TABLE courses
 (
-    id              bigserial       CONSTRAINT courses_pk PRIMARY KEY,
+    id              integer         PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
     uuid            uuid            DEFAULT uuid_generate_v4() NOT NULL,
-    code            varchar         NOT NULL UNIQUE,
+    code            varchar         NOT NULL,
```

Comment:
shouldn't it be unique?, there shouldn't be 2 courses with the same codes

Reply:
Yeah, it should be unique for sure.

Reply:
https://github.com/sumelms/microservice-course/pull/19#discussion_r1520508615

Reply:
I added a unique index based on the code and a "deleted_at" field because I anticipate that we might have duplicate codes as long as one of them has been marked as deleted:

```
CREATE UNIQUE INDEX courses_code_uindex
    ON courses (code, deleted_at) NULLS NOT DISTINCT;
```

Reply:
https://github.com/sumelms/microservice-course/pull/19#discussion_r1520508615

Reply:
This can prevent for us to unarchive the information, I prefer to keep the possibilty, specially if someone archive by mistake.

Reply:
@rluders Can you help us here?

Reply:
yep, It may seem logical, but it is always good to consider what we are determining as a business rule, and not just the user experience. In this format, if someone archives a course, even by mistake, it will not be possible to unarchive it if someone enters the same code, inactivating all relationships and data related to that course. If the user wants to consult the course history, which means consulting archived courses, this could mean a list of courses without code distinctions and without knowing why that course was archived. If we are considering these factors, it is okay to continue like this.

Reply:
we may add this as part of the archive/unarchive behivior discussuion to no continues to block this PR, WDYT?

Reply:
@rluders

Reply:
Agreed.